### ISSUE TITLE
Training the Original Superpoint Network from scratch 

### ISSUE THREAD

**Issue by vanguard478** [eric-yyjau/pytorch-superpoint#98]
Hi, 
Thanks a lot for this amazing repo and also the blog post. 
I am trying to train the original superpoint network [SuperPoint_MagicLeap](https://github.com/magicleap/SuperPointPretrainedNetwork/blob/master/demo_superpoint.py) using the steps highlighted in this repo. I had to make some changes to deal with some issues in the first steps as below :

- In `models/SuperPointNet_pretrained.py`  I changed this [line](https://github.com/eric-yyjau/pytorch-superpoint/blob/5eb75d74df27c07f6e7311df8f167e2a9c01a798/models/SuperPointNet_pretrained.py#L23) to `class SuperPointNet_pretrained(torch.nn.Module)`
-  Also in the [Train_model_heatmap.py](https://github.com/eric-yyjau/pytorch-superpoint/blob/5eb75d74df27c07f6e7311df8f167e2a9c01a798/Train_model_heatmap.py) , I had to make changes to the variable receiving the return of the model as in the modified Superpoint Models, the return type is `dict` where as the original one returns a `tuple`: 
 ```
            if self.config["model"]["name"] == "SuperPointNet_pretrained":
                semi, coarse_desc = outs[0], outs[1]
            else:
                semi, coarse_desc = outs["semi"], outs["desc"]
```

I am using the same config file [magicpoint_shapes_pair.yaml](https://github.com/eric-yyjau/pytorch-superpoint/blob/5eb75d74df27c07f6e7311df8f167e2a9c01a798/configs/magicpoint_shapes_pair.yaml) as the one mentioned in Step 1 of the [README](https://github.com/eric-yyjau/pytorch-superpoint/blob/master/README.md#1-training-magicpoint-on-synthetic-shapes) with the following parameters changed :
```
front_end_model: 'Train_model_heatmap'
model:
    name: 'SuperPointNet_pretrained'
    params: {
    }
    detector_loss:
        loss_type: 'softmax'

    batch_size:  512 # 64
retrain: True # set true for new model
reset_iter: True
#pretrained: ##train from scratch
```
With these changes, I am able to start the training without any issues.  I have also changed the batch size parameter to `batch_size: 512`  and increased the `learning_rate: 0.008`
The training loss from tensorboard is as below and am not sure if the training is going corretcly:
![train-loss](https://user-images.githubusercontent.com/56787203/208097220-a36bdda3-a425-4ee0-ada3-b0a62ef79060.svg)

Is my current changes sufficient for the training of the MagicPoint network using the Original Superpoint Network?
